src/encord_active/lib/common/project.py

```diff
@@ -0,0 +1,245 @@
+        self.project_dir: Path = project_dir
+
+        # read project's metadata and hash
+        self.project_meta: Dict[str, str] = fetch_project_meta(project_dir)
+        self.project_hash: str = self.project_meta["project_hash"]
+
+        # read project ontology
+        ontology_file_path = project_dir / "ontology.json"
+        if not ontology_file_path.exists():
+            raise FileNotFoundError(f"Expected file `ontology.json` at {project_dir}")
+        self.ontology: OntologyStructure = OntologyStructure.from_dict(
+            json.loads((project_dir / "ontology.json").read_text())
+        )
+
+        # read label rows' metadata
+        label_row_meta_file_path = project_dir / "label_row_meta.json"
+        if not label_row_meta_file_path.exists():
+            raise FileNotFoundError(f"Expected file `label_row_meta.json` at {project_dir}")
+        self.label_row_meta: Dict[str, LabelRowMetadata] = {}
+        for lr_hash, lr_meta in itertools.islice(json.loads(label_row_meta_file_path.read_text()).items(), subset_size):
+            self.label_row_meta[lr_hash] = LabelRowMetadata.from_dict(lr_meta)
+
+        # read label rows and their images
+        self.label_rows: Dict[str, LabelRow] = {}
+        self.image_paths: Dict[str, List[Path]] = {}
+        for lr_hash in self.label_row_meta.keys():
+            lr_file_path = project_dir / "data" / lr_hash / "label_row.json"
+            lr_images_dir = project_dir / "data" / lr_hash / "images"
+            if not lr_file_path.exists():  # todo log this issue
+                continue
+            if not lr_images_dir.exists() or not lr_images_dir.is_dir():  # todo log this issue
+                continue
+            self.label_rows[lr_hash] = LabelRow(json.loads(lr_file_path.read_text()))
+            self.image_paths[lr_hash] = list(lr_images_dir.iterdir())
+
+    @classmethod
+    def from_encord_project(cls, project_dir: Path, encord_project: EncordProject):
+        """
+        Construct a Project from Encord platform.
+
+        :param project_dir: The root directory of the project.
+        :param encord_project: Encord project from where the data is fetched.
+        :return:
+        """
+        if not project_dir.exists():
+            raise FileNotFoundError("`project_dir` must point to an existing directory")
+        if not project_dir.is_dir():
+            raise NotADirectoryError("`project_dir` must point to an existing directory")
+
+        # todo enforce clean up when we are sure it won't impact performance in other sections (like PredictionWriter)
+        # also don't forget to add `from shutil import rmtree` at the top (pylint tags it as unused right now)
+        # clean project_dir content
+        # for path in project_dir.iterdir():
+        #     if path.is_file():
+        #         path.unlink()
+        #     elif path.is_dir():
+        #         rmtree(path)
```

Comment:
The plan of this constructor is to actually update everything inside the data folder so we want to overwrite. As it is written now it will do a clean sweep of the folder, including metrics and whatnots, and we surely don't want that but it's a start 😄. Is not very clear what should remain here because metrics cvs files will become obsolete also and in the future we will want to run the metrics only in the data that changed.

To sum up, how to correctly update the data and metrics it's a topic on its own. Also, the value proposition of current project structure is to limit (get rid of) optional parameters like `override=False`.